crates/matrix-sdk/src/widget/machine/mod.rs

```diff
@@ -173,6 +195,42 @@ impl WidgetMachine {
         }
     }
 
+    #[instrument(skip_all, fields(request_id))]
+    fn send_from_widget_response(
+        &self,
+        raw_request: Raw<FromWidgetRequest>,
+        response_data: impl Serialize,
+    ) {
+        let mut object = match raw_request.deserialize_as::<IndexMap<String, Box<RawJsonValue>>>() {
```

Comment:
I wonder if there is a way to write this more elegantly, without us needing to insert fields :) But probably not for now.

Reply:
I can't think of something much better. Anything else would likely be lossy, this takes the request exactly and just adds one field (`response`).